lm/vocabulary.py

```diff
@@ -146,3 +146,47 @@ def run(self):
         with open(self.out_vocabulary.get_path(), "w") as fout:
             for word in sorted(vocabulary):
                 fout.write(f"{word}\n")
+
+
+class VocabularyFromTextJob(Job):
+    """
+    Extract vocabulary from given text files based on frequency.
+    """
+
+    def __init__(self, file_paths, size=1000000, counts=False):
+        """
+        :param List[Path] file_paths: paths to the text files
+        :param int size: expected size of the vocabulary
```

Comment:
I find `size` as a name not that a clear. is this the size the user expects or is this the max limit? then maybe something like `max_num_words`?